cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/cache/watcher_cache.go

```diff
@@ -1004,6 +1006,30 @@ func (c *NamespacedResourceWatcherCache) ForceAndFetch(key string) (interface{},
 	return c.fetch(key)
 }
 
+// force a particular key to be processed
+func (c *NamespacedResourceWatcherCache) ForceIfNotProcessingAndFetch(key string) (interface{}, error) {
+	// There is one use case when the client needs to be able to do an Add(), regardless of whether
+	// the item is being processed, e.g. when the corresponding value goes through several quick changes.
+	// Then there is a separate use case when the client doesn't want to do an Add
+	// if the item is currently being processed, such as a .Get() operation that leads to a lengthy
+	// .Add() cuncurrently with another .Get() immediately. Executing two 2 .Add() operations does
+	// not solve any problems just slows the whole thing down.
```

Comment:
I'm missing a lot of context, so this could be totally irrelevant, but wouldn't the described issue be avoided automatically if the `.Get()` and `.Add()` operations were in [transactions](https://redis.io/docs/manual/transactions/)? (ie. the second `.Get()` operation should not result in another `.Add()`?)

Reply:
We aren't using Redis transactions. Whether or not we should is a different topic. If we do its going to take some of design/work. Currently, redis is used with a (rate-limiting) [queue](https://github.com/kubernetes/client-go/blob/master/util/workqueue/queue.go) which doesn't grok transactions. The `.Get() `about redis cache and the queue, while `.Add()` is all about the queue. I don't see how redis transactions would be relevant yet. Sorry it wasn't clear from the comment.